pkg/manager/member/ticdc_upgrader.go

```diff
@@ -108,7 +108,14 @@ func (u *ticdcUpgrader) Upgrade(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulS
 			}
 			continue
 		}
-		mngerutils.SetUpgradePartition(newSet, i)
+
+		err = gracefulShutdownTiCDC(tc, u.deps.CDCControl, u.deps.PodControl, pod, ordinal, "Upgrade")
+		if err != nil {
+			return err
+		}
```

Comment:
If only one ticdc, will wait until the timeout?

Reply:
No, it skips graceful shutdown, see

https://github.com/pingcap/tidb-operator/blob/6c3bfa34dfdf7df4bfeb2502928c9b1df75d273c/pkg/controller/ticdc_control.go#L53-L65